malloc_mmap.go

```diff
@@ -51,3 +55,16 @@ func putChunk(chunk []byte) {
 	freeChunks = append(freeChunks, p)
 	freeChunksLock.Unlock()
 }
+
+func clearChunks() error {
+	freeChunksLock.Lock()
+	defer freeChunksLock.Unlock()
+	freeChunks = nil
+	for _, data := range baseChunks {
+		baseChunk := data
```

Comment:
What is the point of this line?

Reply:
If you don't call `Munmap` on the slice retrieved from `Mmap`, it will leak. `freeChunks` contains subslices of this slice, which will return an error, so the original slice needs to be stored for book keeping